src/SigSpec.Core/SigSpecGenerator.cs

```diff
@@ -95,38 +96,38 @@ private IEnumerable<MethodInfo> GetChannelMethods(Type type)
             });
         }
 
-        private async Task<SigSpecOperation> GenerateOperationAsync(Type type, MethodInfo method, JsonSchemaGenerator generator, SigSpecSchemaResolver resolver, SigSpecOperationType operationType)
+        private SigSpecOperation GenerateOperation(Type type, MethodInfo method, JsonSchemaGenerator generator, SigSpecSchemaResolver resolver, SigSpecOperationType operationType)
         {
             var operation = new SigSpecOperation
             {
-                Description = await method.GetXmlSummaryAsync(),
+                Description = method.GetXmlDocsSummary(),
                 Type = operationType
             };
 
             foreach (var arg in method.GetParameters())
             {
-                var parameter = await generator.GenerateWithReferenceAndNullabilityAsync<SigSpecParameter>(
-                    arg.ParameterType, arg.GetCustomAttributes(), resolver, async (p, s) =>
+                var parameter = generator.GenerateWithReferenceAndNullability<SigSpecParameter>(
+                    arg.ParameterType.ToContextualType(), resolver, (p, s) =>
                     {
-                        p.Description = await arg.GetXmlDocumentationAsync();
+                        p.Description = arg.GetXmlDocs();
                     });
 
                 operation.Parameters[arg.Name] = parameter;
             }
 
             var returnType =
```

Comment:
Check: This must be a proper contextual type to carry NRT info

Reply:
I fixed one bug here, is this what you were thinking of? https://github.com/RicoSuter/SigSpec/pull/25/commits/793d9195b2d141e762ef52ec0e052e0b9f7c13fb